File: src/main/java/at/wambo/podcaster/model/Bookmark.java
package at.wambo.podcaster.model;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import java.time.Duration;
import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.Id;
import javax.persistence.ManyToOne;
import javax.persistence.Table;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

/**
 * @author Martin 26.06.2017
 */
@Data
@Entity
@Table(name = "bookmarks")
@AllArgsConstructor
@NoArgsConstructor
public class Bookmark {

  @GeneratedValue
  @Id
  private Integer id;

  @Column(nullable = false)
  private Duration position;

  @ManyToOne(targetEntity = User.class, optional = false)
  private User user;

  @ManyToOne(targetEntity = FeedItem.class, optional = false)
  @JsonIgnoreProperties("bookmarks")
  private FeedItem feedItem;
}

File: frontend/src/model/Bookmark.ts
import * as moment from "moment";
import User from "./User";
import * as F from "./FeedItem";
import FeedItem from "./FeedItem";

export default interface Bookmark {
  id: number;
  position: moment.Duration | string;
  user: User;
  feedItem: FeedItem;
}

export function parseDates(bookmark: Bookmark): void {
  if (typeof bookmark.position === "string") {
    bookmark.position = moment.duration(bookmark.position);
  }
  F.parseDates(bookmark.feedItem);
}